lib/Transforms/ConvertSecretWhileToStaticFor/ConvertSecretWhileToStaticFor.cpp

```diff
@@ -0,0 +1,161 @@
+#include "mlir/include/mlir/Analysis/DataFlow/DeadCodeAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlowFramework.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Affine/IR/AffineOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"   // from @llvm-project
+#include "mlir/include/mlir/Dialect/SCF/IR/SCF.h"       // from @llvm-project
+#include "mlir/include/mlir/IR/Diagnostics.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/MLIRContext.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"             // from @llvm-project
+#include "mlir/include/mlir/IR/PatternMatch.h"          // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"                 // from @llvm-project
+#include "mlir/include/mlir/Interfaces/SideEffectInterfaces.h"  // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"           // from @llvm-project
+#include "mlir/include/mlir/Support/LogicalResult.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+
+namespace mlir {
+namespace heir {
+
+#define GEN_PASS_DEF_CONVERTSECRETWHILETOSTATICFOR
+#include "lib/Transforms/ConvertSecretWhileToStaticFor/ConvertSecretWhileToStaticFor.h.inc"
+
+struct SecretWhileToStaticForConversion : OpRewritePattern<scf::WhileOp> {
+  using OpRewritePattern<scf::WhileOp>::OpRewritePattern;
+
+ public:
+  SecretWhileToStaticForConversion(DataFlowSolver *solver, MLIRContext *context)
+      : OpRewritePattern(context), solver(solver){};
+
+  // TODO(#846): Add support for do-while loops
+  LogicalResult matchAndRewrite(scf::WhileOp whileOp,
+                                PatternRewriter &rewriter) const override {
+    auto whileCondition = whileOp.getConditionOp()->getOperand(0);
+
+    auto *conditionSecretnessLattice =
+        solver->lookupState<SecretnessLattice>(whileCondition);
+
+    if (!conditionSecretnessLattice) {
+      InFlightDiagnostic diag = mlir::emitWarning(
+          whileOp.getLoc(),
+          "Secretness for scf.while condition has not been set:");
+      return failure();
+    }
+
+    bool isConditionSecret =
+        conditionSecretnessLattice->getValue().getSecretness();
+
+    // If condition is not secret, no transformation is needed
+    if (!isConditionSecret) {
+      return failure();
+    }
+
+    auto maxIterAttr = whileOp->getAttrOfType<IntegerAttr>("max_iter");
+
+    // If maximum iteration attribute is not provided, emit a warning and return
+    // failure
+    if (!maxIterAttr) {
+      InFlightDiagnostic diag = mlir::emitWarning(
+          whileOp.getLoc(),
+          "Cannot convert secret scf.while to static affine.for "
+          "since a static maximum iteration attribute has not been provided:");
```

Comment:
```suggestion
          "since a static maximum iteration attribute (`max_iter`) has not been provided on the scf.while op:");
```

The error should clearly state the expected attribute name, or else someone who doesn't know about this requirement would not know how to fix it when seeing the error.